### ISSUE TITLE
Extremely quick overfitting and unstable training with YOLO26l

### ISSUE THREAD

**Issue by ZagButNoZig** [ultralytics/ultralytics#23399]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi,

I’m currently training a YOLO model (yolo26l) for nuclei detection. My main dataset contains 828 images. My training setup is:

- Pretrain for 10 epochs on two related datasets (~800 images aswell)
- Fine-tune on my main dataset for 30 epochs

However, I consistently observe overfitting after ~20 epochs: validation mAP starts to drop while the training metrics continue improving.

Is this behavior expected for a dataset of this size?

I’ve tried a variety of anti-overfitting approaches (augmentation changes, regularization, etc.), but so far everything I’ve tried either has no effect or makes performance worse. I also searched through existing issues but couldn’t find anything that directly matches my situation.

My code mostly uses defaults (the only change is the pretraining LR, since 0.001 worked better for me):

```python
def main() -> None:
    args = parse_args()
    model = YOLO(args.weights)

    if args.pretrain_data and args.pretrain_epochs > 0:
        model.train(
            data=args.pretrain_data,
            epochs=args.pretrain_epochs,  # 10
            imgsz=args.imgsz,  # 640
            device=args.device,
            project=args.project,
            name=f"{args.name}_pretrain",
            amp=True,
            optimizer="MuSGD",
            momentum=0.9,
            lr0=0.001,
            close_mosaic=0,
        )

        save_dir = Path(getattr(model.trainer, "save_dir", ""))
        best = save_dir / "weights" / "best.pt"
        last = save_dir / "weights" / "last.pt"

        model = YOLO(str(best))

    model.train(
        data=args.data,
        epochs=args.epochs,  # 30
        imgsz=args.imgsz,  # 640
        device=args.device,
        project=args.project,
        name=args.name,
        amp=True,
    )

if __name__ == "__main__":
    main()
```
Here is the training plot. Validation mAP starts declining after about 20 epochs:
<img width="2400" height="1200" alt="Image" src="https://github.com/user-attachments/assets/8234fd84-4246-4965-a294-52ad76f139de" />

And here is the dataset distribution:
![Image](https://github.com/user-attachments/assets/dc097381-d4ae-44e8-bea6-62b7ee45e5d3)

It may also be relevant that I train with fixed box sizes, since that is how the target dataset is labeled.

I realize it is hard to remote debug this kind of thing, so I am mainly looking for general guidance on whether this is expected and what the most common causes and fixes are for this pattern.

Kind regards

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ZagButNoZig, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for sharing the details, plots, and your training snippet—this helps a lot 🙏📈 This is an automated response, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report (e.g., you suspect something is wrong/unstable in training beyond expected generalization behavior), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. This should include a tiny dataset subset (or synthetic repro), exact command/code to run, and the full logs needed to reproduce the behavior.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). In particular, please add:
-Exact `ultralytics` version, OS, GPU, CUDA, PyTorch versions
-Your full training logs for both pretrain and finetune runs (including hyperparameters that were auto-selected)
-Your dataset YAML (paths + `names`) and a couple of representative labeled images (train/val) showing the “fixed box sizes” labeling pattern
-Confirmation of how your train/val split was created (and whether there’s any leakage across splits)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can try a smaller model. You're training millions of parameters on just 828 images. The model can memorize that.

Try `yolo11s.pt`

**Comment by ghost** [comment]
Thank you for your quick response @Y-T-G !

I tried a few different sizes. All seem to be worse than 26L. I tried: 26x, 26s, 11s, here are the training charts for 11s:

<img width="2400" height="1200" alt="Image" src="https://github.com/user-attachments/assets/077f4fb4-c206-4c04-9568-ad3c96d1e903" />

and for 26s:

<img width="2400" height="1200" alt="Image" src="https://github.com/user-attachments/assets/d30bb4d8-4155-440d-ae9f-def3a30208c4" />

I don't have the training graph for 26x anymore but it was slightly worse than 26L. Interestingly 11s is slightly worse than 26L, but 26s is significantly worse. (Final val: 42.3 vs 41.8 vs 40.4)

**Comment by Y-T-G** [comment]
What imgsz you're using?

**Comment by ghost** [comment]
All training params (except for pretrain LR and epochs) are defaults. So 640. The original images are 1024², but due to vram and in order to align with what yolo was trained on I use the default.

**Comment by Y-T-G** [comment]
Your objects are too small, so you should try higher imgsz

**Comment by ghost** [comment]
<img width="2400" height="1200" alt="Image" src="https://github.com/user-attachments/assets/198613ce-519e-47f7-8129-a00db0b4ec03" />

This is what 11s (others require too much VRAM) looks like with imgsz = 1024 (the original res). I don't think its a resolution issue. I guess I have to keep trying.

**Comment by Y-T-G** [comment]
How many validation images do you have?

**Comment by ghost** [comment]
127

**Comment by glenn-jocher** [comment]
127 val images is on the small side for a stable `mAP50-95` curve, so some of the “drop after ~20 epochs” you’re seeing can be simple metric variance (especially if your 828 images are patches from a smaller number of slides—make sure the split is by slide/patient, not by patch). In practice it’s fine if the best checkpoint is earlier; I’d lean on early stopping (`patience`) and keep the “best epoch” rather than forcing 30 epochs, and for nuclei specifically consider training on tiled crops (e.g., 1024→multiple 640 crops) to preserve small-object detail without needing `imgsz=1024` VRAM. See our notes on [overfitting and common mitigations](https://www.ultralytics.com/glossary/overfitting) and the [evaluation/fine-tuning guidance (incl. small-object tiling)](https://docs.ultralytics.com/guides/model-evaluation-insights/).

```python
model.train(
    data=args.data,
    imgsz=640,
    epochs=200,      # allow training, let early stop pick the best
    patience=15,     # stop when val no longer improves
    dropout=0.05,    # light regularization (optional)
)
```

**Comment by ghost** [comment]
Thanks for the advice! I have tried it with patience 50, but still observe overfitting consistently.

(yolo11s)

<img width="2400" height="1200" alt="Image" src="https://github.com/user-attachments/assets/65698c5c-6823-4733-9098-0e5e7509f76d" />

**Comment by glenn-jocher** [comment]
`patience` only takes effect when `epochs > patience` (e.g. `patience=50` won’t early-stop a 30-epoch run), so first double-check that mismatch; if you’re already training long enough and still see a consistent peak then the next knob I’d turn is fine-tuning LR (for a small-ish dataset it’s common to need a much smaller `lr0` than defaults to avoid quickly memorizing). See our [model training tips (epochs + early stopping)](https://docs.ultralytics.com/guides/model-training-tips/) for the intended behavior.

```python
model.train(
    data=args.data,
    imgsz=640,
    epochs=300,
    patience=50,
    lr0=1e-4,     # smaller fine-tune step
    dropout=0.05  # optional
)
```
